cli/flox-rust-sdk/src/providers/publish.rs

```diff
@@ -200,60 +206,61 @@ pub fn check_build_metadata(
     })
 }
 
-fn gather_build_repo_meta(environment: &PathEnvironment) -> Result<LockedUrlInfo, PublishError> {
+fn gather_build_repo_meta(environment: &impl Environment) -> Result<LockedUrlInfo, PublishError> {
     // Gather build repo info
     let git = match environment.parent_path() {
         Ok(env_path) => GitCommandProvider::discover(env_path)
-            .map_err(|e| PublishError::UnsupportEnvironmentState(Box::new(e)))?,
-        Err(e) => return Err(PublishError::UnsupportEnvironmentState(Box::new(e))),
+            .map_err(|e| PublishError::UnsupportEnvironmentState(format!("Git error {e}")))?,
+        Err(e) => return Err(PublishError::UnsupportEnvironmentState(e.to_string())),
     };
 
     let origin = git
         .get_origin()
-        .map_err(|e| PublishError::UnsupportEnvironmentState(Box::new(e)))?;
+        .map_err(|e| PublishError::UnsupportEnvironmentState(format!("Git error {e}")))?;
 
     let rev = origin
         .revision
         .ok_or(PublishError::UnsupportEnvironmentState(
-            "No revision found".to_string().into(),
+            "No revision found".to_string(),
         ))?;
 
     let rev_count = git
         .rev_count(rev.as_str())
-        .map_err(|e| PublishError::UnsupportEnvironmentState(Box::new(e)))?;
+        .map_err(|e| PublishError::UnsupportEnvironmentState(format!("Git error {e}")))?;
+
+    let rev_date = git
+        .rev_date(rev.as_str())
+        .map_err(|e| PublishError::UnsupportEnvironmentState(format!("Git error {e}")))?;
 
     Ok(LockedUrlInfo {
         url: origin.url,
         rev,
         rev_count,
-        rev_date: None,
+        rev_date,
     })
 }
 
 fn gather_base_repo_meta(
     flox: &Flox,
-    environment: &PathEnvironment,
+    environment: &mut impl Environment,
 ) -> Result<Option<LockedUrlInfo>, PublishError> {
```

Comment:
Can this be `Result<LockedUrlInfo, PublishError>` rather than `Result<Option<LockedUrlInfo>, PublishError>` now that we either return something or an error, until we allow no base catalog dependency again?

This part drew my attention to it because I was wondering why it would ever be `None` after `check_environment_metadata()`: https://github.com/flox/flox/pull/2324/files#diff-afe9925d59eecc04a1c6a9b929a6fc7861bb67e2cd20edb486f93a9661f24c73R152-R157

Reply:
Probably a good idea!